core/src/tangram.h

```diff
@@ -166,6 +166,9 @@ void runOnMainLoop(std::function<void()> _task);
 // Run this task asynchronously to Tangram's main update loop.
 void runAsyncTask(std::function<void()> _task);
 
+// Capture a snapshot of the current frame
+std::vector<unsigned int> captureSnapshot();
```

Comment:
Yes I agree, I'm in favor for changing that for just an allocated pointer.
Something like this would make sense to me:
```
int width = Tangram::getViewportWidth();
int height = Tangram::getViewportHeight();
// [...] alloc on snapshot by width/height
Tangram::captureSnapshot(snapshot, &scale); // write on allocated unsigned int*
```

The Java API/iOS can ensure the large enough buffer since we store can store the width and height in `MapController`/`ViewController`.